mat64/dense_arithmetic.go

```diff
@@ -351,6 +351,7 @@ func (m *Dense) Dot(b Matrix) float64 {
 }
 
 // Mul takes the matrix product of a and b, placing the result in the receiver.
+// If a or b are also the receiver, a workspace will be allocated.
```

Comment:
I think that in conjunction with "temporary" and the absence of any handle to a workspace clients should not over-read the word "workspace".

If we implement pools, this all becomes moot - which to me is a good reason for doing that.